src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -36,13 +37,17 @@
     private static final String LINE_INSERTED_SYMBOL = "+";
     private static final String STARTING_LINE_NUMBER_GROUP_NAME = "startingLineNumber";
     private static final String FILE_CHANGED_GROUP_NAME = "filePath";
+    private static final String MATCH_GROUP_FAIL_MESSAGE_FORMAT = "Failed to match the %s group for:\n%s";
 
     private static final int LINE_CHANGED_HEADER_INDEX = 0;
 
     private static final Pattern STARTING_LINE_NUMBER_PATTERN = Pattern.compile(
             "-(\\d)+(,)?(\\d)* \\+(?<startingLineNumber>\\d+)(,)?(\\d)* @@");
     private static final Pattern FILE_CHANGED_PATTERN = Pattern.compile("\n(\\+){3} b/(?<filePath>.*)\n");
 
+    private static final Predicate<String> NEW_EMPTY_FILE_PREDICATE = Pattern.compile("^new file mode 100644\n"
+            + "index [a-zA-Z0-9]{7}\\.{2}[a-zA-Z0-9]{7}$").asPredicate();
```

Comment:
`\\.` seems like an additional `\` here.
But anyway, let's make use of the `+++` and `---` functionality and simplify this
-> `^new file mode [a-zA-Z0-9\n\. ]*$`

Reply:
>\\. seems like an additional \ here.

There is a need for 2 `\`, since the 1st `\` is meant to tell that the 2nd `\` is the actual backslash rather than an escape character.
Anyway, within the square brackets, there is no need for `\` :sweat_smile:.

Reply:
I see, I am used to using online regex test so I kinda forgot the syntax for java.